generator/src/framework_sources.rs

```diff
@@ -205,12 +204,20 @@ export function parseTypeName(name: Type): { typeName: string; typeArgs: Type[]
   return { typeName: parsed.name, typeArgs: parsed.params as string[] }
 }
 
+export function isTransactionArgument(arg: GenericArg): arg is TransactionArgument {
+  if (!arg || typeof arg !== 'object' || Array.isArray(arg)) {
+    return false
+  }
+
+  return 'kind' in arg
+}
+
 export function obj(txb: TransactionBlock, arg: ObjectArg) {
-  return is(arg, TransactionArgument) ? arg : txb.object(arg)
+  return isTransactionArgument(arg) ? arg : txb.object(arg)
 }
```

Comment:
@hayes-mysten can `TransactionArgument` be exported as value here https://github.com/MystenLabs/sui/blob/main/sdk/typescript/src/builder/export.ts#L9-L10 or `isTransactionArgument` be added to the SDK so we don't have to resort to this?

Reply:
We are trying to move away from the `superstruct` definitions in the SDK.  To me I think this belongs here rather than in the SDK.  The goal here is to discriminate between the types in the GenericArg union, which is a concept that belongs to sui-client-gen.

in this case, we know that the arg is one of the types supported by  `GenericArg`, which is a constraint we couldn't have as easily in the SDK.  We would need to resort back to matching the way that `superstruct` does (basically checking that all the fields of the object match) which is a very expensive way to handle this.

Reply:
Ok looks good. Thanks!